File: samples/src/main/java/net/ickis/deluge/samples/SimpleClient.java
package net.ickis.deluge.samples;

import io.reactivex.Single;
import net.ickis.deluge.api.DelugeJavaClient;
import net.ickis.deluge.api.Torrent;

import java.util.Optional;
import java.util.function.Function;

@SuppressWarnings("unused")
public class SimpleClient {
    public static final String IP = "localhost";
    public static final int PORT = 3333;
    public static final String USERNAME = "username";
    public static final String PASSWORD = "password";
    public static final String MAGNET_LINK = "";

    @SuppressWarnings("ResultOfMethodCallIgnored")
    public static void main(String[] args) {
        try (DelugeJavaClient client = new DelugeJavaClient(IP, PORT, USERNAME, PASSWORD)) {
            Single<Optional<String>> torrentIdSingle = client.addTorrent(MAGNET_LINK);
            String torrentId = torrentIdSingle.blockingGet().orElseThrow(() -> new IllegalArgumentException("Bad magnet link"));
            Single<Torrent> torrentStatus = client.getTorrentStatus(torrentId);
            client.removeTorrent(torrentId, true).blockingGet();
        }
    }

    static <T> T create(Function<DelugeJavaClient, T> function) {
        try (DelugeJavaClient client = new DelugeJavaClient(IP, PORT, USERNAME, PASSWORD)) {
            return function.apply(client);
        }
    }
}

File: samples/src/main/kotlin/net/ickis/deluge/samples/SimpleClient.kt
package net.ickis.deluge.samples

import kotlinx.coroutines.runBlocking
import net.ickis.deluge.api.*

const val IP = "localhost"
const val PORT = 1234
const val USERNAME = "username"
const val PASSWORD = "password"
const val MAGNET_LINK = ""

@Suppress("UNUSED_VARIABLE")
fun main() {
    runBlocking {
        delugeClient(IP, PORT, USERNAME, PASSWORD) { client ->
            val torrentId: String = client.addTorrent(MAGNET_LINK) ?: throw IllegalArgumentException("Bad magnet link")
            val torrent: Torrent = client.getTorrentStatus(torrentId)
            val removed: Boolean? = client.removeTorrent(torrentId, true)
        }
    }
}

fun <T> simpleClient(block: suspend (DefaultDelugeClient) -> T) = runBlocking {
    delugeClient(IP, PORT, USERNAME, PASSWORD, block)
}
